code/modules/mob/hear_say.dm

```diff
@@ -167,7 +167,22 @@
 
 /mob/proc/hear_sleep(var/message)
 	var/heard = ""
+	var/opentag = 0
+	var/closetag = 0
 	if(prob(15))
+		while(1) //Remove the HTML formatting so people don't hear colour
```

Comment:
After talking it over with @tigercat2000 and they're fairly sure `strip_html_properly` will work.